articles/javascript-iterations-which-one-is-fast/index.md

```diff
@@ -0,0 +1,657 @@
+
+***For example:***
+```js
+for (let loop = 0; loop < 6; loop++) {
+  console.log("hello world")
+}
+```
+There we have it, two lines of code give us a perfect six hello world as we anticipated. Shorter code means faster development time, code readability, and lesser performance overheads.
+
+JavaScript has different kinds of iterations statements called loops. They include `for`, `while`, `do while`, `for in`, `for of`, and `for each`. They all do the same job, to repeat an action several times, but have different ways to start and end a loop.ster? When it comes to loops, the most significant questions are which loop to use? Which loop is fastest?
+
+In this guide, we will discuss these loops with simple and basic examples, compare and test the execution performance of each loop.
+
+### For Loop
+For loop is the commonly used loop. If you are not new to the world of programming, you might have come across a for loop statement under other programming languages such as `java`, `c++`, `c`, etc. A for statement loops through a specified or fixed number of elements until the specified condition is met.
+
+For statement consist of the following parts:
+```js
+for (initializer; condition; increment){
+    statement //loop body
+}
+```
+- Initialization expression - it is a variable declaration that initializes a loop counter `loop = 0`.
+- Condition expression - for loop executes when the condition value is true, and if it is false, the loop terminates when `loop < 5`. It is a test condition that determines the number of times a statement will iterate.
+- Increment expression - `loop++` updates the loop by increasing or decreasing the loop counter.
+- Statement - `console.log ("hello world")` if the condition executes to true, the loop statement is executed until the condition expression `loop < 5` is false.
+
+***For loop example:***
+```js
+for (let loop = 0; loop < 5; loop++){
+console.log("hello world") //loop body
+}
+```
+
+### Do While
+Repeats a statement until the specified condition evaluates to false.
+```js
+do {
+  statement; //loop body
+}  while (condition);
+```
+The statement is checked/execute before the condition. If the condition is true, the statement loop will be executed again. In the do-while loop, the next statement executes before the condition rechecks the previously executed statement. If the condition is true, the statement is executed again until the condition evaluates to false.
+```js
+let i = 0;
+do {
+console.log("hello world");
+  i ++;
+} while (i < 5);
+```
+In the above example, `do` iterates before `while` then `i` reiterates until the condition executes to false. I.e. until the loop cycle is no longer less than five `(i < 5)`.
```

Comment:
This statement is unclear and technically `i` does not iterate and is just a variable at this point.